packages/graphback-datasync/src/providers/DeltaProvider.ts

```diff
@@ -0,0 +1,203 @@
+import { applyIndexes } from "@graphback/runtime-mongo";
+import { GraphQLObjectType } from 'graphql';
+import { Db, ObjectId, FilterQuery } from "mongodb"
+import { getTableName, GraphbackOperationType } from '@graphback/core';
+
+function getDeltaTableName(baseType: GraphQLObjectType) {
+  return `Graphback-${getTableName(baseType)}-Delta`
+}
+
+type DeltaOps = GraphbackOperationType.CREATE | GraphbackOperationType.UPDATE | GraphbackOperationType.DELETE;
+
+export interface ConflictingFieldsReport {
+  _id: ObjectId,
+  conflicts: {
+    [fieldName: string]: any
+  },
+  version: number
+}
+
+interface DeltaDocument {
+  docId: ObjectId,
+  op: DeltaOps,
+  timestamp: Date,
+  version: number,
+  sets: {
+    [fieldName: string]: any
+  }
+}
+
+
+/**
+ * Provides an interface for interacting with deltas and stuff
+ */
+export class DeltaProvider {
+  protected collectionName: string;
+  protected db: Db;
+
+  public constructor(baseType: GraphQLObjectType, db: Db) {
+    this.db = db;
+
+    this.collectionName = getDeltaTableName(baseType);
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    applyIndexes([
+      {
+        key: {
+          "docId": 1,
+          "version": 1
+        }
+      }
+    ], this.db.collection(this.collectionName));
+  }
+
+
+  public async insertDiff(_id: string|ObjectId, ts: string|number, documentVersion: number, updateData: any, op: DeltaOps) {
+    const docId = getObjectIdFromString(_id);
+    const timestamp = getDateFromTimestamp(ts);
+    const sets = getDiffEntry(updateData);
+
+    const deltaDocument: DeltaDocument = {
+      op,
+      docId,
+      timestamp,
+      version: documentVersion,
+      sets
+    }
+
+    const res = await this.db.collection(this.collectionName).insertOne(deltaDocument);
+
+    return res.insertedId;
+  }
+
+  public async sync(lastSync: number, filter?: FilterQuery<any>) {
+    const syncPipeline = [
+      {
+        $match: {
+          timestamp: {
+            $gte: new Date(lastSync)
+          }
+
+        },
+      },
+      {
+        $group: {
+          _id: "$docId",
+          sets: {
+            $mergeObjects: "$sets"
+          },
+          lastOp: {
+            $last: "$op"
+          },
+          lastUpdated: {
+            $last: "$timestamp"
+          }
+        }
+      }, {
+        $replaceRoot: {
+          newRoot: {
+            $mergeObjects: [
+              {
+                _id: "$_id",
+                _deleted: { $eq: ["$lastOp", "delete"] },
+                updatedAt: {
+                  $toLong: "$lastUpdated"
+                }
+              },
+              "$sets"
+            ]
+          }
+        }
+      }, {
+        $match: filter
+      }];
+
+    return this.db.collection(this.collectionName).aggregate(syncPipeline).toArray()
+  }
+
+  public async checkForConflicts(_id: string|ObjectId, version: number, updateData: any): Promise<ConflictingFieldsReport> {
```

Comment:
Another missunderstanding.

Conflict detection doesn't need delta tables. 
Implementation bellow doesn't make sense to me. 
Let's chat.

We need delta to handle conflict but we can detect it based on o the source table.

Reply:
This is only checking if the fields that are being modified by a client have changed since the client fetched the document, this happens when version is mismatched in the source table, continuing: if the fields have been modified it throws conflict error, if the fields have not changed, it accepts the update

Reply:
In other words this is conflict being resolved, conflict checking is occurring in the data provider

Reply:
oh.. So this is in wrong place then. As mentioned above we should wrap anything that suggest what fields or operations affect detection. Please check current implementation.